File: 234-palindrome-linked-list/palindrome-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public bool IsPalindrome(ListNode head) {
        ListNode slow=head,fast=head;
        Stack<int> st = new Stack<int> ();
        while(fast!=null && fast.next!=null){
            st.Push(slow.val);
            slow=slow.next;
            fast=fast.next.next;
        }
        if(fast!=null){
            slow=slow.next;
        }

        while(slow!=null && st.Peek()==slow.val){
            slow=slow.next;
            st.Pop();
        }
        return st.Count==0;
    }
}

File: 234-palindrome-linked-list/palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPal(vector<int> vec){
        for(int i=0;i<vec.size();i++){
            if(vec[i]!=vec[vec.size()-1-i]) return false;
        }
        return true;
    }
    bool isPalindrome(ListNode* head) {
        vector<int> vec;
        ListNode* temp=head;
        while(temp){
            vec.push_back(temp->val);
            temp=temp->next;
        }
        return isPal(vec);
    }
};